config/env_spec.go

```diff
@@ -278,6 +281,9 @@ type APIOperation struct {
 	// Completely overrides the API (proxy) level settings. No merging will happen.
 	ContextVariables []ContextVariable `yaml:"context_variables,omitempty" mapstructure:"context_variables,omitempty"`
 
+	// DynamicMetadata to be given to Envoy. If specified, overrides any DynamicMetadata specified at the API level.
+	DynamicMetadata map[string]interface{} `yaml:"dynamic_metadata,omitempty" mapstructure:"dynamic_metadata,omitempty"`
```

Comment:
Do we have any use cases that will have something other than strings? If not, should we restrict this map[string]string first?

Reply:
Makes sense. Thanks.